src/Admin/IsAdherentDatagrid.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace AppBundle\Admin;
+
+use Sonata\AdminBundle\Datagrid\DatagridInterface;
+
+class IsAdherentDatagrid extends DatagridDecorator
+{
+    private $manager;
+
+    public function __construct(DatagridInterface $decorated, IsAdherentManager $manager)
+    {
+        parent::__construct($decorated);
+
+        $this->manager = $manager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getResults()
+    {
+        static $results = null;
+
+        if (null === $results) {
+            $results = $this->decorated->getResults();
+
+            foreach ($results as $result) {
+                $result->isAdherent = $this->manager->isAdherent($result->getEmailAddress());
```

Comment:
IMO we should use a better way to avoid the X queries with this foreach

Reply:
We discused about that with @ottaviano last friday and we agree that this shall be reworked together.

Reply:
It was solved by @ottaviano.